src/test/java/mate/academy/security/AuthenticationServiceImplTest.java

```diff
@@ -0,0 +1,85 @@
+package mate.academy.security;
+
+import static org.mockito.ArgumentMatchers.any;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.RoleService;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class AuthenticationServiceImplTest {
+    private static final String VALID_EMAIL = "bob@i.ua";
+    private static final String VALID_PASSWORD = "123456";
+    private AuthenticationService authenticationService;
+    private UserService userService;
+    private RoleService roleService;
+    private PasswordEncoder passwordEncoder;
+    private User expected;
+
+    @BeforeEach
+    void setUp() {
+        userService = Mockito.mock(UserService.class);
+        roleService = Mockito.mock(RoleService.class);
+        passwordEncoder = Mockito.mock(PasswordEncoder.class);
+
+        authenticationService = new AuthenticationServiceImpl(
+                userService, roleService, passwordEncoder);
+
+        expected = new User();

```

Comment:
There is no need to reinstantiate objects/mocks for each tests unless you wish to reset some state (ususally applicable for dao).
Please recheck other classes

```suggestion
class AuthenticationServiceImplTest {
    private static final String VALID_EMAIL = "bob@i.ua";
    private static final String VALID_PASSWORD = "123456";
    
    private UserService userService = mock(UserService.class);
    private RoleService roleService = mock(RoleService.class);
    private PasswordEncoder passwordEncoder = mock(PasswordEncoder.class);
    private AuthenticationService authenticationService = new AuthenticationServiceImpl(
                userService, roleService, passwordEncoder);

    private User expected;

    @BeforeEach
    void setUp() {
         expected = new User();
```